CMakeLists.txt

```diff
@@ -34,9 +34,9 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 if(NOT WIN32)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -g -DDEBUG")
```

Comment:
MSVC just ignores them if not used and compiles tests correctly, so I wouldn't add boilerplate to solve a problem that doesn't exist.  
At the end of the day, maintainability is a good thing. ;-). 
Moreover, `WIN32` doesn't mean MSVC, so this change is wrong in any case. Are you experiencing an actual problem or is the change a premature optimization? In the first case, what error are you encountering?